### PR TITLE
agol del col, new resource indicator

### PR DESCRIPTION
**Pull request by garyqi-to**
One update to ckan_resource operator is to add is_new indicator for branching for backup after get_or_create resource, simplify the flow control.

Another update to agol operator is to add the capability to remove more columns (param delete_col in lower case), so for report type of extract we can get rid of all the null geometry etc.

Also added agol-datastore dag generator and updated police annual report update

### CODE REVIEW

**Review by mackeynichols — COMMENTED** [review]
Test review comment

File: dags/datasets/agol-datastore.py
Comment (garyqi-to) [reply dags/datasets/agol-datastore.py:131]: I'm trying again: The function to check the return result is needed for branching

File: dags/datasets/agol-datastore.py
Comment (mackeynichols) [reply dags/datasets/agol-datastore.py:48]: ok fair - I think we'll take the `yaml` approach **later** for AGOL (and other) resources, since it will be a lot of work 👍

File: dags/datasets/agol-datastore.py
Comment (mackeynichols) [reply dags/datasets/agol-datastore.py:1]: ok, sure. 
Let's add these to the list of DAGs that could be refactored when we look to configure jobs in YAMLs (like we mention here: https://github.com/open-data-toronto/operations/pull/38#discussion_r742112374)

File: dags/datasets/agol-datastore.py
Comment (mackeynichols) [reply dags/datasets/agol-datastore.py:131]: Ah, I see. You use that to determine whether you need to backup an existing resource or create a new one. Looks good to me.

A thought: I wonder if we could make an operator to make that process simpler? Would it even be worth is?
 Like - an operator that says "if theres a resource already, Ill back it up, otherwise Ill create a new one"

This comment is nothing you need to address in this PR @garyqi-to dont worry - Im just thinking out loud

File: dags/datasets/agol-datastore.py
Comment (mackeynichols) [reply dags/datasets/agol-datastore.py:170]: I disagree - it would be good and it would save us work as we go on to create many DAGs that will repeat this process. It prevents us from writing redundant code.

That being said, I wont let that stand in the way of us merging this PR - it's something we'll have to consider when we look to configure DAGs with `yaml`

**Review by mackeynichols — APPROVED** [review]
Thanks for addressing these changed @garyqi-to 
I'll wait for your green light before I merge this into `main` and we turn these jobs on in PROD

We'll also keep the conversation going about larger, yaml/xls/csv configured DAGs, which addresses some of the issues we discussed in this PR.

Thanks for your work here dude

**Pull request merged**
